R/coh_plot.R

```diff
@@ -0,0 +1,324 @@
+  plt <- ggplot2::ggplot() +
+    ggplot2::geom_ribbon(
+      ggplot2::aes(x = results_1$time,
+        ymin = results_1$plot_lower, ymax = results_1$plot_upper
+      ),
+      fill = colors[1], alpha = 0.2
+    ) +
+    ggplot2::geom_step(
+      ggplot2::aes(x = results_1$time, y = results_1$plot,
+        color = "c1"
+      )
+    ) +
+    ggplot2::geom_ribbon(
+      ggplot2::aes(x = results_2$time,
+        ymin = results_2$plot_lower, ymax = results_2$plot_upper
+      ),
+      fill = colors[2], alpha = 0.2
+    ) +
+    ggplot2::geom_step(
+      ggplot2::aes(x = results_2$time, y = results_2$plot,
+        color = "c2"
+      )
+    ) + {
+    if (percentage) {
+      ggplot2::scale_y_continuous(labels = scales::percent)
+    }
+  } +
+    ggplot2::scale_color_manual(name = "Vaccine Status",
+      values = colors,
+      labels = vacc_status
+    ) +
+    ggplot2::theme_classic() + {
+    if (cumulative) {
+      ggplot2::ylab("Cumulative hazard")
+    } else {
+      ggplot2::ylab("Survival probability")
+    }
+  } +
+    ggplot2::xlab("Time to event (Days)") +
+    ggplot2::labs(colour = "Vaccine Status") +
+    ggplot2::theme(axis.text = ggplot2::element_text(size = 13),
+                   axis.title = ggplot2::element_text(size = 15),
+                   legend.title = ggplot2::element_text(size = 15),
+                   legend.text = ggplot2::element_text(size = 13))
+  return(plt)
+}
+
+#' Function to plot the vaccine coverage in a cohort
+#'
+#' The function returns a plot of the vaccine coverage or the cumulative
+#' coverage (if cumulative = TRUE).
+#' The return is a 2-axis ggplot2 element with the number of vaccines per date
+#' in the left-axes and the coverage per date in the right-axes.
+#' It is possible to manipulate the colors, labels, legend and most of
+#' the graphic elements.To do so follow the convention:
+#' c("c1" = "color1", "c2" = "color2")
+#'
+#' @inheritParams coh_coverage
+#' @param doses_count_color color assigned to the doses count
+#' @param coverage_color color assigned to the coverage calculation
+#' @param cumulative if TRUE returns the cumulative number of doses
+#' over the time window
+#' @return 2-axis ggplot2 plot of vaccine coverage and daily doses
+#' @examples
+#' data("cohortdata")
+#' start_cohort <- as.Date("2044-01-01")
+#' end_cohort <- as.Date("2044-12-31")
+#' date_interval <- c(start_cohort, end_cohort)
+#' plot_coverage(
+#'   data = cohortdata,
+#'   vacc_date_col = "vaccine_date_1",
+#'   unit = "month",
+#'   doses_count_color = "steelblue",
+#'   coverage_color = "mediumpurple",
+#'   date_interval = date_interval,
+#'   cumulative = FALSE
+#' )
+#' @export
+plot_coverage <- function(data,
+                          vacc_date_col,
+                          unit = c("day", "month", "year"),
+                          doses_count_color = "steelblue",
+                          coverage_color = "mediumpurple",
```

Comment:
Here too, I would simply pick a good default and let users overwrite it later if they wish.

Reply:
I will keep the definition of colors for now to avoid changes in the user test that we'll do next week. I created this issue to cover this in the next pull request https://github.com/epiverse-trace/vaccineff/issues/32#issue-1927852696